### ISSUE TITLE
Gradient problem when the number of devices is 1

### ISSUE THREAD

**Issue by SivilTaram** [jzhang38/TinyLlama#88]
First thanks for all of the efforts you have done for the TinyLlama project - it's awesome!

Recently I found a spurious problem. When there is only one card, the training gradient would disappear. Is it expected? Is the model training on one card currently not supported?

Thanks!

**Comment by jzhang38** [comment]
Can you share more details including the error message?

**Comment by radarFudan** [comment]
I ran the script on a single node using the following command. The training loss will not descent. The evaluated gradient norm is zero. 

```
lightning run model \
    --node-rank=0  \
    --accelerator=cuda \
    --devices=1 \
    --num-nodes=1 \
    pretrain/tinyllama.py --devices 1 --train_data_dir data/slim_star  --val_data_dir data/slim_star
```
    
Gradient norm:
```
if not is_accumulating:
  total_norm = torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=grad_clip)
  # Print the gradient norm
  print("Gradient norm: ", total_norm)
  
  fabric.clip_gradients(model, optimizer, max_norm=grad_clip)
```

**Comment by jzhang38** [comment]
https://github.com/Lightning-AI/lit-gpt/issues/689
probably an issue with lightning fabrc, or some toggle needs to be on.

**Comment by ChaosCodes** [comment]
Hi, if you want you train your code with 1 gpu, you perhaps need to set `empty_init=Flase` [here](https://github.com/jzhang38/TinyLlama/blob/main/pretrain/tinyllama.py#L135). But I have no idea yet why it will lead to the gradient problem.
